src/sql/workbench/contrib/notebook/browser/notebookActions.ts

```diff
@@ -375,9 +375,13 @@ export class RunAllCellsAction extends Action {
 	}
 	public override async run(context: URI): Promise<void> {
 		try {
-			this._telemetryService.sendActionEvent(TelemetryKeys.TelemetryView.Notebook, TelemetryKeys.NbTelemetryAction.RunAll);
 			const editor = this._notebookService.findNotebookEditor(context);
 			await editor.runAllCells();
+
+			const azdata_notebook_guid: string = editor.model.getMetaValue('azdata_notebook_guid');
```

Comment:
If you do this then the events won't send events when the run fails. 

Please fix the test instead (I'm assuming it's something with the telemetry service not being set up correctly before the test)